fix: run with --scope to allow perf to trace the benchmark process

not-matthias opened the pull request:
Unfortunately when running with --scope, we can't use `EnvironmentFile` which is why I had to add it to the sh cmd. Perf can now trace the benchmark process again.

GuillaumeLagrange reviewed (changes requested):
--scope actually inherits env from what I have see

```
export MY_VAR="hello"

systemd-run env | grep MY_VAR

systemd-run --scope env | grep MY_VAR
```

not-matthias commented:
> --scope actually inherits env from what I have see
> 
> ```
> export MY_VAR="hello"
> 
> systemd-run env | grep MY_VAR
> 
> systemd-run --scope env | grep MY_VAR
> ```

Doesn't work on my machine. Seems to only inherit the system env: 
```bash
> export THIS_IS_A_TEST="ZZZZZZZZZZZZZZZZZZZZZZZZZZZZZ"
> echo $THIS_IS_A_TEST
ZZZZZZZZZZZZZZZZZZZZZZZZZZZZZ
> sudo systemd-run --scope -- env | grep THIS_IS_A_TEST
Running as unit: run-p2129159-i2129160.scope; invocation ID: fdf21a15d503409891f434dd43243ad1
```

art049 commented on src/run/runner/wall_time/executor.rs:
```diff
@@ -34,7 +34,11 @@ impl WallTimeExecutor {
-            .map(|(env, value)| format!("{env}={value}"))
+            .map(|(env, value)| {
+                // Properly quote the value to handle special characters
+                let escaped_value = value.replace("\\", "\\\\").replace("\"", "\\\"");
+                format!("export {env}=\"{escaped_value}\"")
```
ouch creating a file we need to source like this seems like we can introduce a ton of annoying edge cases

not-matthias replied:
We could also use the `printenv` cmd and write it to the file directly.

art049 commented on src/run/runner/wall_time/perf/mod.rs:
```diff
@@ -251,7 +251,10 @@ impl PerfRunner {
             .await?
             .into_iter()
-            .filter_map(Result::ok)
+            .filter_map(|result| {
+                debug!("Copy task result: {result:?}");
+                result.ok()
+            })
```
should we keep this?

art049 reviewed (changes requested):
Is using source the only way to inherit the env?

not-matthias commented on Cargo.toml:
```diff
@@ -54,6 +54,7 @@ debugid = "0.8.0"
 memmap2 = "0.9.5"
 nix = { version = "0.29.0", features = ["fs", "user"] }
 futures = "0.3.31"
+shell-quote = "0.7.2"
```
I was initially using this for escaping the runner args, but now we are only using it to create the script to validate the env var: 
```rust
fn env_var_validation_script(env: &str, expected: &str) -> String {
    let expected: String = expected.quoted(Bash);
    format!(
        r#"
if [ "${env}" != {expected} ]; then
  echo "FAIL: Environment variable not set correctly"
  echo "Got: '${env}'"
  exit 1
fi
"#
    )
}
```

Will move it to dev-deps though :+1:

not-matthias commented:
Added `rstest_reuse` and moved the constants with the test scripts/env vars into the array. 

Note: the last 2 commits should be ignored, as they are merged from https://github.com/CodSpeedHQ/runner/pull/99 (not sure why Github didn't recognize that, I'll close both once this PR is merged)

art049 reviewed (approved):
Nice! @GuillaumeLagrange I let you do the final review

GuillaumeLagrange reviewed (approved):
lgtm to me!
